flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/windowing/TopSpeedWindowing.scala

```diff
@@ -72,7 +74,34 @@ object TopSpeedWindowing {
       } else {
         println("Executing TopSpeedWindowing example with default inputs data set.")
         println("Use --input to specify file input.")
-        env.fromCollection(genCarStream())
+        env.addSource(new SourceFunction[CarEvent]() {
+
+          val speeds = Array.fill[Integer](numOfCars)(50)
+          val distances = Array.fill[Double](numOfCars)(0d)
+          @Transient lazy val rand = new Random()
+
+          var counter = 0
+          var isRunning:Boolean = true
+
+          override def run(ctx: SourceContext[CarEvent]) = {
+            while (isRunning && counter < numOfEvents) {
```

Comment:
I think we shouldn't limit the number of emitted events. It is a streaming example and as such it should continuously run.